src/constitutive_relations/eos/carbon_decomposition_rate_evaluator.cc

```diff
@@ -56,19 +54,29 @@ CarbonDecomposeRateEvaluator::Evaluate_(const State& S, const std::vector<Compos
 {
   Tag tag = my_keys_.front().second;
   Epetra_MultiVector& res_c = *result[0]->ViewComponent("cell", false);
+  Epetra_MultiVector& dz_c = *result[0]->ViewComponent("cell", false);
```

Comment:
This will not work in future code, where getting two views of the same vector will throw.

Don't bother getting a temporary vector here.  Just change the second loop:

```
for (int col = 0; col != mesh.columns.num_columns_owned; ++col) {
    const auto& col_cells = mesh.columns.getCells(col);
   for (int i = 0; i != col_cells.size(); ++i) {
      double dz = mesh.getCellVolume(col_cells[i]) / mesh_surf.getCellVolume(col);
      double f_temp = Func_Temp(temp_c[0][col_cells[i]], q10_);
      double f_depth = Func_Depth(depth_c[0][col_cells[i]]);
      double f_pres_temp = Func_TempPres(temp_c[0][col_cells[i]], pres_c[0][col_cells[i]]);
      res_c[0][c] = (f_temp * f_depth * f_pres_temp * dz_c[0][c]) * (1 - por_c[0][col_cells[i]]);
   }
}
```